File: src/CommandLineCommands/ConfigCommands/ConfigRemoveCommand.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

class ConfigRemoveCommand : ConfigBaseCommand
{
    public string? Key { get; set; }
    public string? Value { get; set; }

    public override string GetCommandName()
    {
        return "config remove";
    }

    public List<Task<int>> Execute(bool interactive)
    {
        var tasks = new List<Task<int>>();
        tasks.Add(Task.FromResult(ExecuteRemove(Key, Value, Scope ?? ConfigFileScope.Local, ConfigFileName)));
        return tasks;
    }

    private int ExecuteRemove(string? key, string? value, ConfigFileScope scope, string? configFileName)
    {
        ConsoleHelpers.WriteDebugLine($"ExecuteRemove; key: {key}; value: {value}; scope: {scope}; configFileName: {configFileName}");
        if (string.IsNullOrWhiteSpace(key))
        {
            throw new CommandLineException($"Error: No key specified.");
        }

        if (value == null)
        {
            throw new CommandLineException($"Error: No value specified.");
        }

        var isFileNameScope = scope == ConfigFileScope.FileName && !string.IsNullOrEmpty(configFileName);
        if (isFileNameScope) _configStore.RemoveFromList(key, value, configFileName!);
        if (!isFileNameScope) _configStore.RemoveFromList(key, value, scope, true);

        var listValue = isFileNameScope
            ? _configStore.GetFromFileName(key, configFileName!).AsList()
            : _configStore.GetFromScope(key, scope).AsList();

        ConsoleHelpers.WriteLine(listValue.Count > 0
            ? $"{key}:\n" + $"  - {string.Join("\n  - ", listValue)}"
            : $"{key}: (empty list)",
            overrideQuiet: true);

        return 0;
    }
}

File: ts/src/CommandLineCommands/ConfigCommands/ConfigRemoveCommand.ts
import { ConfigBaseCommand } from './ConfigBaseCommand';
import { ConfigFileScope } from '../../Configuration/ConfigFileScope';
import { KnownSettings } from '../../Configuration/KnownSettings';
import { ConsoleHelpers } from '../../Helpers/ConsoleHelpers';

/**
 * Command to remove a value from a list configuration setting.
 */
export class ConfigRemoveCommand extends ConfigBaseCommand {
  public key?: string;
  public value?: string;

  getCommandName(): string {
    return 'config remove';
  }

  isEmpty(): boolean {
    return !this.key || this.key.trim() === '' || this.value === undefined;
  }

  async executeAsync(_interactive: boolean): Promise<number> {
    return await this.executeRemove(
      this.key!,
      this.value!,
      this.scope ?? ConfigFileScope.Local
    );
  }

  private async executeRemove(
    key: string,
    value: string,
    scope: ConfigFileScope
  ): Promise<number> {
    if (!key || key.trim() === '') {
      throw new Error('Error: No key specified.');
    }
    if (value === undefined) {
      throw new Error('Error: No value specified.');
    }

    // Normalize the key if it's a known setting
    let normalizedKey = key;
    if (KnownSettings.isKnown(key)) {
      normalizedKey = KnownSettings.getCanonicalForm(key);
    }

    await this._configStore.removeFromList(normalizedKey, value, scope);

    // Display the updated list in YAML format
    const updatedValue = await this._configStore.getFromScope(normalizedKey, scope);
    if (updatedValue && Array.isArray(updatedValue.value) && updatedValue.value.length > 0) {
      ConsoleHelpers.writeLine(`${normalizedKey}:`);
      for (const item of updatedValue.value) {
        ConsoleHelpers.writeLine(`- ${item}`);
      }
    } else {
      ConsoleHelpers.writeLine(`${normalizedKey}: (empty)`);
    }

    return 0;
  }

}